### ISSUE TITLE
Component wrapped by `<PrimeReactProvider>` runs twice in the production

### ISSUE THREAD

**Issue by ubau** [primefaces/primereact#4823]
### Describe the bug

`<App>` component wrapped by `<PrimeReactProvider>` runs twice in the production if some configuration options are used, e.g.:
```ts
  const { setRipple } = useContext(PrimeReactContext);

  setRipple(true);
```

I catch this by reading cookies and clearing them after in my <App> component. Because of running twice on the second run they are empty and cannot be read. 

### Reproducer

https://codesandbox.io/s/primereact-demo-forked-r6zksp

### PrimeReact version

9.6.2, 9.6.1

### React version

18.x

### Language

TypeScript

### Build / Runtime

Vite

### Expected behavior

I expect wrapped component to run only once, as it did before with now deprecated `PrimeReact.ripple = true;`

It would be nice to remove deprecation from `PrimeReact` - it is much simpler and easier way to set options and it runs faster as well because there are no provider, context and hooks involved.

**Comment by ubau** [comment]
Yes, it runs twice in production mode.

As I said before I read some cookies, use their values and after that delete these cookies. On second run cookies are empty (checked with `console.log("value:", value);` - in console I can see two rows - first with value and second empty ). Of course I can check if they have some values and use them only then. 
Yes, I use `<React.StrictMode>` but problem is that my `<App>` component now runs twice in production mode which is not acceptable. 

With now deprecated `PrimeReact.ripple = true;` it runs only once as expected.

**Comment by melloware** [comment]
I totally agree it is an issue. Can you update my sandbox here to show the two console logs so we can add the test case here: https://codesandbox.io/s/primereact-demo-forked-r6zksp?file=/src/App.js

**Comment by ubau** [comment]
Hmm, I provide these cookies from Wildfly app server from which I also run my React SPA.

Here is just an example how I use it:
```js
export const App = () => {
  const { setRipple  } = useContext(PrimeReactContext);
  setRipple(true);

  const cookie = Cookies.get("JSMV") || "";
  const value = window.atob(cookie);

  console.log("value:", value);

  if (!!value) {
    const dispatch = useAppDispatch();
    dispatch(setMyValue(value));
  }

  Cookies.remove("JSMV", { path: "/fin", sameSite: "Lax" });

  return (
    <>
      <Header />
      <MainView />
      <Footer />
      <ConfirmDialog />
    </>
  );
};
```
`useAppDispatch()` is from Redux Toolkit (RTK).

When I login in my app I see in console two rows: 
value: xyz
value:

but there must be only one - value: xyz

If I just comment out `setRipple(true);` `App` runs only once and there is only one row - value: xyz

My app is in production mode.

**Comment by melloware** [comment]
OK see this ticket: https://github.com/primefaces/primereact/issues/4811

I think their instuctions are misleading can you try what I have in there wiht the `useMount` hook?

**Comment by ubau** [comment]
Nope, it still runs twice.

**Comment by melloware** [comment]
OK thanks for testing

**Comment by melloware** [comment]
OK added reproducer to your ticket: https://codesandbox.io/s/primereact-demo-forked-r6zksp?file=/src/App.js

**Comment by ubau** [comment]
It would be nice to have old way `PrimeReact.ripple = true;` back.

**Comment by melloware** [comment]
I am working on something to see if it works..

**Comment by melloware** [comment]
OK two things...

1. In my demo if I turn Strict.Mode off I only see 1 console log?
2. In 10.0.0 which I will try and backport to 9.7.0 you can do this..

```xml
<PrimeReactProvider value={{ ripple: true }}>
```

Where you can pass the initial values for your app so the hook never is needed or fires if you truly want Ripple for your whole app.

**Comment by ubau** [comment]
Ok, I will try 9.7.0 when it's ready but removing `<React.StrictMode>` is not a solution.

`<PrimeReactProvider value={{ ripple: true }}>` looks better than now, but it sill requires one additional provider and context for whole app just for couple options. What was the reason to deprecate `PrimeReact`, if I may ask.

**Comment by melloware** [comment]
I have no idea why PrimeTek did it i think it related to solving a problem with static variables?  make things more React "like" ?
